### PR TITLE
Move app directory to root, delete api

### PR DESCRIPTION
**Pull request by roryf**
# Description

Having created new repo [`Volunteer-api`](https://github.com/Scottish-Tech-Army/Volunteer-api), this change removes the `api` directory and moves `app` directory up to root, so the repo only contains code for app front-end.

There should be no changes to behaviour or how the app itself is developed, except that you no longer need to `cd` into the app directory and the api should be run from other repo.

I tried to update readmes as best I could to remove api-specific docs and anywhere that mentioned the `app` directory. Other than those, all the changes here are deletions of api files or renames of app files.

This will cause major conflicts with #203 and any currently active branches, suggest we prioritise getting this merged asap so make those easier.

# Testing locally

Checkout the branch, `npm install` from root and follow your usual dev process for running the app, check everything works the same.

This will require cloning the api repo and running it locally from there.

# Checklist

- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have removed any unnecessary comments or console logging
- [] I have made corresponding changes to the documentation (if required)
- [] I have addressed accessibility, if needed
- [] I have followed best practices, e.g. NativeBase approaches and theming
- [] I have checked the app in dark mode, if making front-end design changes
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] I have updated the version numbers in `package.json` files in the [app](DEPLOYMENT.md#app-deployment) and/or [api](DEPLOYMENT.md#api-deployment-on-aws) directories as needed

### CODE REVIEW

**Comment by CatMac228** [comment]
Hey @roryf! The link above to the new repo isn't working for me - guessing we might to be given access to it?

**Comment by roryf** [comment]
Sorry @CatMac228 thought I'd added the Volunteer app team to the repo, but appears you're not in that! Will see if we can fix that, in the meantime added you to the other repo.

File: .expo/settings.json
Comment (DJMF2015) [inline .expo/settings.json:1]: Out of curiosity what is the purpose of this?

Comment (roryf) [reply .expo/settings.json:1]: Good question, I have no idea! There is a readme in the folder, which has this little nugget:

> Should I commit the ".expo" folder? No, you should not share the ".expo" folder

Think I'll leave as is for now so this is a clearer changeset, but sounds like we should delete it and add to gitignore

Comment (DJMF2015) [reply .expo/settings.json:1]: Sure - makes sense!

**Review by DJMF2015 — APPROVED** [review]
@roryf sorry for the delay reviewing. I have gone through the changes and tried locally on my machine. The changes made are useful meaning the folder structure of the repos are more streamline the previously. Just one quick question: i notice that there is still an "api" folder. But is the idea to remove this once the other repo containing the api folder you setup is completed?  The application itself ran smoothly when i tried locally. One issue i did get when run the test suites was the error _'Test environment jest-environment-jsdom cannot be found. Make sure the testEnvironment configuration option points to an existing node module.'_ Although if i install the missing module, most tests pass there was one failing test - though the error message made it difficult to tease the cause '_TypeError: Cannot redefine property: performance'_. I don't think it's a big deal and likely an issue at my end. That aside - all seemed to be fine and good and thanks for doing.

**Comment by CatMac228** [comment]
Hey @roryf! 

This might be an issue with my build but when I run locally the app bundles okay but then crashes with these errors  

```
ERROR  TypeError: Cannot read property 'executionEnvironment' of undefined, js engine: hermes
 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes
```

Not seeing any errors when running on the main branch - have you come across these before?

**Comment by roryf** [comment]
@CatMac228 thats an interesting one! Just to double check, are you running it from the root of the repo not the app folder? That folder might still be there locally, since there are git-ignored files in it. Once merged everyone will need to delete the app and api directories locally to avoid any confusion (+ @DJMF2015 suspect that's what you're seeing too)

If not that, I'm at a bit of a loss…

**Comment by CatMac228** [comment]
Got it working! A bit extreme but deleted and re-cloned the repo, had to install this package @expo/ngrok@^4.1.0 and then the App loaded 🎆 I am seeing these warnings which I wasn't seeing before...I may have missed updates needed to the .env file


```
 WARN  ❗  You are running in development mode but you are using the production API server -- this is usually a bad idea, you should be using your local API development server instead. You probably need to stop the app, add a STA_API_BASE_URL value to your .env file, and then restart it. 
    at App (http://ziwjwvy.anonymous.19000.exp.direct/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false:101522:44)
    at withDevTools(App) (http://ziwjwvy.anonymous.19000.exp.direct/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false:101187:27)
    at RCTView
    at View (http://ziwjwvy.anonymous.19000.exp.direct/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false:48633:43)
    at RCTView
    at View (http://ziwjwvy.anonymous.19000.exp.direct/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false:48633:43)
    at AppContainer (http://ziwjwvy.anonymous.19000.exp.direct/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false:48510:36)
    at main(RootComponent) (http://ziwjwvy.anonymous.19000.exp.direct/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false:85001:28)
 LOG  🛈 Using API version: v1
 LOG  No ID token in storage getIdToken
 LOG  No ID token found getCurrentUser
```

**Review by CatMac228 — APPROVED** [review]
Approved :)

**Review by PammyLo — APPROVED** [review]
Works as expected, thanks @roryf!

**Pull request merged**